src/AbstractLinesBeforeNamespaceFixer.php

```diff
@@ -34,22 +35,66 @@
      */
     protected function fixLinesBeforeNamespace(Tokens $tokens, $index, $expected)
     {
-        // if we've got a <?php, then subtracted the number of new lines it
-        // contains from the expected number in the following whitespace
-        if (isset($tokens[$index - 2])) {
-            $opening = $tokens[$index - 2];
-            if ($opening->isGivenKind(T_OPEN_TAG)) {
-                $expected -= substr_count($opening->getContent(), "\n");
+        // Let's determine the total numbers of new lines before the namespace
+        // and the opening token
+        $precedingNewlinesTotal = 0;
+        $precedingNewlinesInOpening = 0;
+        $openingToken = null;
+        for ($i = 1; $i <= 2; ++$i) {
+            if (isset($tokens[$index - $i])) {
+                $token = $tokens[$index - $i];
+                if ($token->isGivenKind(T_OPEN_TAG)) {
+                    $openingToken = $token;
+                    $precedingNewlinesInOpening = substr_count($token->getContent(), "\n");
+                    $precedingNewlinesTotal += $precedingNewlinesInOpening;
+                    break;
+                }
+                if (false === $token->isGivenKind(T_WHITESPACE)) {
+                    break;
+                }
+                $precedingNewlinesTotal += substr_count($token->getContent(), "\n");
             }
         }
 
-        $previousIndex = $index - 1;
-        $previous = $tokens[$previousIndex];
-        if ($previous->isWhitespace()) {
+        if ($expected !== $precedingNewlinesTotal) {
+            $previousIndex = $index - 1;
+            $previous = $tokens[$previousIndex];
             if (0 === $expected) {
-                $tokens->clearAt($previousIndex);
-            } elseif (substr_count($previous->getContent(), "\n") !== $expected) {
-                $tokens[$previousIndex] = new Token(array(T_WHITESPACE, str_repeat("\n", $expected)));
+                // Remove all the previous new lines
+                if ($previous->isWhitespace()) {
+                    $tokens->clearAt($previousIndex);
+                }
+                // Remove new lines in opening token
+                if (0 < $precedingNewlinesInOpening) {
+                    $openingToken->setContent(rtrim($openingToken->getContent()).' ');
```

Comment:
Don't use `$token->setContent(...)` as it's been deprecated on higher branches. Please replace it with `$tokens[$index] = new Token(...)`.